app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    dict__ = dict()
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        nonlocal dict__
+
+        if not dict__.get((*args, inner.__name__,)) is None:
+            print("Getting from cache")
+            return dict__.get((*args, inner.__name__,))
+        print("Calculating new result")
+
+        dict__[(*args, inner.__name__,)] = func(*args)
+        return dict__[(*args, inner.__name__,)]
```

Comment:
the double return statement in the your solution is redundant and can be simplified. The `return dict__[(*args, inner.__name__,)]` statement can be simplified to reduce the number of dictionary lookups and make the code more readable.